gjson.go

```diff
@@ -1405,6 +1407,15 @@ func Get(json, path string) Result {
 	return c.value
 }
 
+// GetReader searches json for the specified path.
+// If working with io.Reader, this method preferred over Get(string(data), path)
```

Comment:
Although this method will save some typing it doesn't make the code faster and more efficient. I'd still merge it but ideally it needs to start reading the stream and as soon it reaches the desired result it should return.
Imagine that you have a huge JSON response. For example:
```
{
     "value": 5,
     // 1MB of other JSON fields here
}
```

If you are buffering you'd still have to read the whole JSON before you can get the value. On the other hand, if you are streaming you'll return as soon as you've found the desired value. Now imagine that you are processing 1 billion events an hour : )
This is what I mean by more efficient code...